connection.js

```diff
@@ -459,6 +459,65 @@ class Connection {
         }
         /******** end of BOARDSIZES *********/
 
+        if (notification.handicap === -1 && config.noautohandicap) {
+            conn_log("no autohandicap");
+            return { reject: true, msg: "For easier bot management, automatic handicap is disabled on this bot, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones" };
+	}
+
+        if (notification.handicap === -1 && config.noautohandicapranked && notification.ranked) {
+            conn_log("no autohandicap for ranked games");
+            return { reject: true, msg: "For easier bot management, automatic handicap is disabled for ranked games on this bot, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones" };
+	}
+
+        if (notification.handicap === -1 && config.noautohandicapunranked && !notification.ranked) {
+            conn_log("no autohandicap for unranked games");
+            return { reject: true, msg: "For easier bot management, automatic handicap is disabled for unranked games on this bot, please manually select the number of handicap stones you want in -custom handicap-, for example 2 handicap stones" };
+	}
```

Comment:
How about:

```
function checkRankedUnranked(option, notification, cb) {
  if (config[option]) {
    return cb("");
  }
  if (config[option+'ranked'] && notification.ranked) {
    return cb(" for ranked games");
  }
  if (config[option+'unranked'] && !notification.ranked) {
    return cb(" for unranked games");
  }
  return nil;
}

if (notification.handicap === -1) {
  let reject = checkRankedUnranked("noautohandicap", notification, (rur) => {
    conn_log("no auto handicap${rur}");
    return { reject: true, msg: "Automatic handicap is disabled${rur} on this bot" };
  });
  if (reject !== nil) {
    return reject;
  }
}
```

Reply:
the first line would need to be `if (config[option] && !config[option+"ranked"] && !config[option+"unranked"]) {`

this idea is interesting, but in some cases we need extra conditions, but yes we can start thinking about this and it would make code a lot more efficient

for the function name, more like `checkFamilyRejects` maybe

(and we dont need to pass into parameter `notification.` i think, isnt it a global variable ?)

and i feel like theres an easier way to write the reject function

and also would be done in a separate PR too @windo 

but yes definitely something i started to think about recently, and would be great to do :+1: